CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+
+We use the following developer-specific tools for documentation. 
+
+* [Sphinx 5.0 or above](https://www.sphinx-doc.org/en/master/) 
+* [nbsphinx](https://nbsphinx.readthedocs.io/en/0.8.9/) 
+* [pydata-sphinx-theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/) 
+
+
+## Builds
+
+We use [Gnu make](https://www.gnu.org/software/make/) for builds.  If you have installed [CmdStan](https://mc-stan.org/users/interfaces/cmdstan) as required for the source to work, then you will already have Gnu make and a sufficient C++ compiler.
+
+
+## Language specific practices
+
+### C and C++ development
+
+* We require the following dependency for C++ development:
+
+* We use the C++1y standard for compilation (`-std=c++1y` in both clang and gcc).  This is partway between C++11 and C++14, and is what Stan requires.
+
+* We try to write standards-compliant code that does not depend on features of specific platforms (except where needed for compatibility).
+
+* We try to follow the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html), but (a) we allow C++ exceptions, and (b) we allow reference arguments.
+
+* We will use [GoogleTest](https://google.github.io/googletest/) for C++ unit testing.
+
+* We recommend using [Clang format](https://clang.llvm.org/docs/ClangFormat.html) with our config file [`.clang-format`](https://github.com/roualdes/bridgestan/blob/main/.clang-format).
+
+
+### Python development
+
+* Python development relies on the external dependency:
+    * [pytest](https://docs.pytest.org/en/7.1.x/) 
+    * [NumPy](https://numpy.org/)
+
+* We integrate a C wrapper of Stan's model class using the [ctypes](https://docs.python.org/3/library/ctypes.html) foreign function interface.
+
+* We use the [numpy.testing](https://numpy.org/doc/stable/reference/routines.testing.html) unit testing framework and run the tests by calling [pytest](https://docs.pytest.org/en/7.1.x/) from the top-level Python directory `bridgestan/python`.
+
+* We autoformat code with [black](https://black.readthedocs.io/en/stable/).
+
+* We recommend but do not require checking the code style and semantics with [PyLint](https://www.pylint.org) using [BridgeStan .pylintrc](https://github.com/roualdes/bridgestan/blob/main/.pylintrc).
+
+### R development
+
+* R dependencies beyond those included in this repo:
+    * [R6 package](https://cran.r-project.org/web/packages/R6/index.html) for reference classes
+    * [testthat](https://testthat.r-lib.org) for unit testing
+
+### Julia development
+
+* Julia dependencies beyond those included in this repo:
+    * [Test](https://docs.julialang.org/en/v1/stdlib/Test/)
+    * [LinearAlgebra](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/)
+
```

Comment:
If you want to link to the relevant Julia docs on the C FFI, they are https://docs.julialang.org/en/v1/base/c/ and https://docs.julialang.org/en/v1/stdlib/Libdl/